### PR TITLE
Add new actions to get sales, reports and downloads reviews from App Store Connect API

### PR DESCRIPTION
**Pull request by r-pedraza**
### PR's key points
Added the actions to download reports and sales and reviews.

I have also added several models that represent the data structure and reduce the complexity by being able to handle the data more easily.

### How to review this PR?
Talk with me.

### CODE REVIEW

File: lib/fastlane/plugin/mm_toolkit/actions/download_reviews_from_app_store_connect.rb [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        end
+  
+        def self.is_supported?(platform)
+          [:ios, :mac, :android].include?(platform)
+        end
+  
+        def self.category
+          # Check https://github.com/fastlane/fastlane/blob/0d1aa50045d57975d8b9e5d5f1f489d82ee0f437/fastlane/lib/fastlane/action.rb#L6
+          # for available categories
+          :source_control
```
Comment (adriangl) [inline lib/fastlane/plugin/mm_toolkit/actions/download_reviews_from_app_store_connect.rb:123]: Set this to
```suggestion
          :app_store_connect
```

File: lib/fastlane/plugin/mm_toolkit/actions/download_sales_and_reports_from_app_store_connect.rb [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        end
+  
+        def self.is_supported?(platform)
+          [:ios, :mac, :android].include?(platform)
+        end
+  
+        def self.category
+          # Check https://github.com/fastlane/fastlane/blob/0d1aa50045d57975d8b9e5d5f1f489d82ee0f437/fastlane/lib/fastlane/action.rb#L6
+          # for available categories
+          :source_control
```
Comment (adriangl) [inline lib/fastlane/plugin/mm_toolkit/actions/download_sales_and_reports_from_app_store_connect.rb:116]: Set this to
```suggestion
          :app_store_connect
```

File: lib/fastlane/plugin/mm_toolkit/helper/app_store_ax_connector/app_store_connect_api.rb [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+  "?filter[frequency]=WEEKLY&filter[reportDate]=#{date}&filter[reportSubType]=SUMMARY&filter[reportType]=SALES&filter[vendorNumber]=#{@vendor_number}&filter[version]=#{app_version}"
+end
+
+def sales_reports_daily_query_params
+  "?filter[frequency]=DAILY&filter[reportSubType]=SUMMARY&filter[reportType]=SALES&filter[vendorNumber]=#{@vendor_number}"
+end
+
+def sales_path_absolute_url
+  base_url+sales_reports_path+sales_reports_query_params
+end
```
Comment (adriangl) [inline lib/fastlane/plugin/mm_toolkit/helper/app_store_ax_connector/app_store_connect_api.rb:67]: Might be worth condensing this into:
```ruby
require 'uri' # at the start of the file

def sales_reports_url(date)
  path = "salesReports"
  query_params = {
    "filter[frequency]" => "DAILY",
    "filter[reportDate]" => date,
    "filter[reportType]" => "SALES",
    "filter[reportSubType]" => "SUMMARY",
    "filter[vendorNumber]" => "#{@vendor_number}"
  }
  query = URI.encode_www_form(query_params)

  URI::HTTP.build(host: base_url, path: path, query: query)
end
```

Give it a try! ;)

File: lib/fastlane/plugin/mm_toolkit/helper/app_store_ax_connector/reviews.rb [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    attr_reader :data, :links, :meta
+
+    def initialize(data, links, meta)
+         @data = Review.from_json(data)
+         @links = links
+         @meta = meta
+    end
+    
+    def self.from_json(file)
+        puts file
```
Comment (adriangl) [inline lib/fastlane/plugin/mm_toolkit/helper/app_store_ax_connector/reviews.rb:13]:
```suggestion
```

**Review by adriangl — CHANGES_REQUESTED** [review]
There are a few things missing, apart from the comments:
- You have to run `rubocop` in order to lint the code. Run:
  ```shell
  bundle exec rubocop -A
  ```
  So the code gets auto-corrected, and fix the missing uncorrected issues
- You have to bump the plug-in version at https://github.com/masmovil/fastlane-plugin-mm_toolkit/blob/master/lib/fastlane/plugin/mm_toolkit/version.rb so consumers can pick up the proper version.

File: lib/fastlane/plugin/mm_toolkit/actions/download_reviews_from_app_store_connect.rb [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+          begin  
+            app_store_connect_account = AppStoreConnectAccount.new(issuer_id, key_id, private_key, vendor_number)
+            app_store_connect_api = AppStoreConnectAPI.new(app_store_connect_account)
+            reviews = app_store_connect_api.get_reviews(app_id, filter_by_date)
+
+            UI.success("Reviews fetched!")
+            reviews
+            puts reviews.collect { |review|
+              puts review.rows
+            }
```
Comment (adriangl) [inline lib/fastlane/plugin/mm_toolkit/actions/download_reviews_from_app_store_connect.rb:51]: Again, you won't need this code 😛 
```suggestion
```

File: lib/fastlane/plugin/mm_toolkit/actions/download_reviews_from_app_store_connect.rb [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+        begin
+          app_store_connect_account = AppStoreConnectAccount.new(issuer_id, key_id, private_key, vendor_number)
+          app_store_connect_api = AppStoreConnectAPI.new(app_store_connect_account)
+          reviews = app_store_connect_api.get_reviews(app_id, filter_by_date)
+
+          UI.success("Reviews fetched!")
+          reviews
+          puts reviews.collect { |review|
+            puts review.rows
+          }
```
Comment (adriangl) [inline lib/fastlane/plugin/mm_toolkit/actions/download_reviews_from_app_store_connect.rb:51]: Don't forget removing this code!!
```suggestion
```

**Review by adriangl — CHANGES_REQUESTED** [review]

**Review by adriangl — DISMISSED** [review]

**Review by adriangl — APPROVED** [review]

**Review by josemanuelmartin — APPROVED** [review]

**Review by sebastianvarela — APPROVED** [review]

**Review by adriangl — APPROVED** [review]

**Pull request merged**